app/models/item.rb

```diff
@@ -9,4 +9,18 @@ class Item < ApplicationRecord
   validates :type, :price, :currency, :external_id, presence: true
   validates :price, numericality: { greater_than: 0 }
   validates :external_id, uniqueness: true
+
+  scope :by_brands, ->(brands) { brands.blank? ? all : where(brands.map { |brand| "brand ILIKE '%#{brand}%'" }.join(' OR ')) }
+  scope :by_model, ->(model) { model.blank? ? all : where('model ILIKE ?', "%#{model}%") }
+  scope :min_max_price, ->(min_price, max_price) { where('price > ? AND price < ?', min_price, max_price) }
+  scope :by_dial_color, ->(dial_color) { dial_color.blank? ? all : where('dial_color ILIKE ?', "%#{dial_color}%") }
+  scope :by_case_material, ->(case_material) { case_material.blank? ? all : where('case_material ILIKE ?', "%#{case_material}%") }
+  scope :by_case_dimensions, ->(case_dimensions) { case_dimensions.blank? ? all : where('case_dimensions ILIKE ?', "%#{case_dimensions}%") }
+  scope :by_bracelet_material, ->(bracelet_material) { bracelet_material.blank? ? all : where('bracelet_material ILIKE ?', "%#{bracelet_material}%") }
+  scope :by_movement_type, ->(movement_type) { movement_type.blank? ? all : where('movement_type ILIKE ?', "%#{movement_type}%") }
```

Comment:
can we predefine possible values for `movement_type`? Or one of any other parameter?